homeassistant/components/insteon/__init__.py

```diff
@@ -78,12 +76,40 @@ async def async_setup_platforms(hass, config):
             add_on_off_event_device(hass, device)
 
     _LOGGER.debug("Insteon device count: %s", len(devices))
-    register_new_device_callback(hass, config)
+    register_new_device_callback(hass)
     async_register_services(hass)
 
+    conf_data = {**config_entry.data}
+    if conf_data.get(CONF_HOST):
+        connection = f"{conf_data[CONF_HOST]}:{conf_data[CONF_PORT]}"
+
+        if conf_data[CONF_HUB_VERSION] == 2:
+            connection = f"http://{connection}"
+        else:
+            connection = f"socket://{connection}"
+    else:
+        connection = conf_data[CONF_DEVICE]
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device_registry.async_get_or_create(
+        config_entry_id=config_entry.entry_id,
+        connections={(CONNECTION_NETWORK_MAC, connection)},
```

Comment:
connection doesn't look like a mac address.

You only need `identifiers` or `connections`